broker/src/main/java/org/apache/rocketmq/broker/client/ProducerManager.java

```diff
@@ -185,4 +190,33 @@ public void unregisterProducer(final String group, final ClientChannelInfo clien
             log.error("", e);
         }
     }
+
+    public Channel getAvaliableChannel(String groupId) {
+        HashMap<Channel, ClientChannelInfo> channelClientChannelInfoHashMap = groupChannelTable.get(groupId);
+        List<Channel> channelList = new ArrayList<Channel>();
+        if (channelClientChannelInfoHashMap != null) {
+            for (Channel channel : channelClientChannelInfoHashMap.keySet()) {
+                channelList.add(channel);
+            }
+            int size = channelList.size();
+            if (0 == size) {
+                log.warn("Channel list is empty. groupId={}", groupId);
+                return null;
+            }
+
+            int index = positiveAtomicCounter.incrementAndGet() % size;
+            Channel channel = channelList.get(index);
+            int count = 0;
+            boolean isOk = channel.isActive() && channel.isWritable();
+            while (isOk && count++ < 3) {
```

Comment:
3 is a magic number

Reply:
I will use a constant to represent this value, which represents the number of retries for fetch an available channel.